staging/src/k8s.io/legacy-cloud-providers/gce/gce_disks.go

```diff
@@ -661,79 +705,79 @@ func (g *Cloud) BulkDisksAreAttached(diskByNodes map[types.NodeName][]string) (m
 // size, in the specified zone. It stores specified tags encoded in
 // JSON in Description field.
 func (g *Cloud) CreateDisk(
-	name string, diskType string, zone string, sizeGb int64, tags map[string]string) error {
+	name string, diskType string, zone string, sizeGb int64, tags map[string]string) (*Disk, error) {
 	// Do not allow creation of PDs in zones that are do not have nodes. Such PDs
 	// are not currently usable.
 	curZones, err := g.GetAllCurrentZones()
 	if err != nil {
-		return err
+		return nil, err
 	}
 	if !curZones.Has(zone) {
-		return fmt.Errorf("kubernetes does not have a node in zone %q", zone)
+		return nil, fmt.Errorf("kubernetes does not have a node in zone %q", zone)
 	}
 
 	tagsStr, err := g.encodeDiskTags(tags)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	diskType, err = getDiskType(diskType)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	mc := newDiskMetricContextZonal("create", g.region, zone)
 
-	err = g.manager.CreateDiskOnCloudProvider(
+	disk, err := g.manager.CreateDiskOnCloudProvider(
 		name, sizeGb, tagsStr, diskType, zone)
 
 	mc.Observe(err)
 	if isGCEError(err, "alreadyExists") {
```

Comment:
this isn't touched by your PR but doesn't feel like idiomatic go and is kind of hard to read because of that. This should all be wrapped in a `if err != nil`, and the final return should be `return disk, nil`

Reply:
Done